src/App.tsx

```diff
@@ -59,7 +59,7 @@ const App = () => {
     }
 
     updateVersion(monaco, version)
-  }, [version, monaco])
+  }, [version, monaco, isLoading])
```

Comment:
Pay attention to this. Here you're executing this effect infinitely now because you're not providing a way for it to be stopped.

I already noticed this report from biome and wanted to fix it myself because the logic as well as the useEffect usage style needs improvements.
A quick fix would be to just remove the `isLoading` from the check, leaving just `if(!monaco) return`, this works because it is essentially what is already happening. Without the isLoading in the deps array, it's always false, so it's like not having the check at all.

The useEffect usage here is correct in the sense that you want to reuse the logic to switch zod version both for when the user manually clicks the version button, and for when a specific version is needed from a shared url or a version stored in the local storage.

Let's reason about this. If we inspect the useEffect:

<img width="530" alt="image" src="https://github.com/user-attachments/assets/b0445eb7-54dd-4481-8b74-65413c980b28">

we see the logic is executed: 

1. at each mount of the component
2. when monaco changes
3. when the version changes

Let's focus on 2 and 3 of course.

2. can be tricky to understand when the monaco object changes because it's an external library, but we can assume it never changes unless we manually call some particular method. This is confirmed by this (which stores the monaco object into a state that should retain a stable identity: https://giacomocerquone.com/blog/whats-an-object-identity-in-javascript/) https://github.com/suren-atoyan/monaco-react/blob/master/src/hooks/useMonaco/index.ts and by the fact that if you create a useeffect that only listens to the monaco object, you see it never retriggers while using the app.

3. the version on the other hand is simple, it can only change from the button handler once the app started (we're not taking into account initial version values though, this is of no interest to us in this regard). So if we're also disabling the very same button when isLoading is true, we are able to also remove the check

Reply:
Now even though what I said should be correct, the thing is that the reason behind that check were more than right. We shouldn't in any case retrigger the effect if the new version is loading.

How would I fix that?

You have to provide a way to retrieve the current zod version after the user has requested a new version to be set so to add an exit condition like this:

```
useEffect(() => {
  if ((isLoading && zod.getCurrentVersion === version) || !monaco) return

  async function updateVersion(monaco: Monaco, ver: string) {
    setIsLoading(true)
    await zod.setVersion(ver)
    await setMonacoDeclarationTypes(monaco, ver)
    setIsLoading(false)
  }

  updateVersion(monaco, version)
}, [version, monaco])
```

zod.getCurrentVersion is made up and if you feel like this can be easily achieved, I'd prefer this honestly

Reply:
It can be done, however since currently the version can change with button click only and very likely remain like this in the future (YAGNI) I'd prefer just revert this and silence the warning for now.
What do you think?

Reply:
Nope, I provided a thorough explanation. Just remove the check